airflow/utils/log/non_caching_file_handler.py

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import os
+
+
+class NonCachingFileHandler(logging.FileHandler):
+    """
+    This is an extension of the python FileHandler that advises the Kernel to not cache the file
+    in PageCache when it is written. While there is nothing wrong with such cache (it will be cleaned
+    when memory is needed), it causes ever-growing memory usage when scheduler is running as it keeps
+    on writing new log files and the files are not rotated later on. This might lead to confusion
+    for our users, who are monitoring memory usage of Scheduler - without realising that it is
+    harmless and expected in this case.
+
+    See https://github.com/apache/airflow/issues/14924
+
+    Adding the advice to Kernel might help with not generating the cache memory growth in the first place.
+    """
+
+    def _open(self):
+        fd = logging.FileHandler._open(self)
+        os.posix_fadvise(fd, 0, 0, os.POSIX_FADV_DONTNEED)
```

Comment:
I wonder if we need to add any special behaviour or check if the advice is available. The https://docs.python.org/3/library/os.html#os.posix_fadvise mentions that posix_fadvise is only available on Unix (I guess that excludes both MacOS and Windows, but I guess it's going to be harmless and ignored if not available. Anyone can check it on MacOS ?

Reply:
I fixed retrieving a wrong file descriptor (looked at wrong open method) and I added try/except to handle any kind of failure when we try to advise the kernel that we do not need to cache the file.